metals/src/main/scala/scala/meta/metals/Linter.scala

```diff
@@ -51,40 +67,63 @@ class Linter(cwd: AbsolutePath) extends LazyLogging {
       )
     )
 
+  private val config: () => Either[String, Option[Input]] =
+    configuration
+      .focus(_.scalafix.confPath)
+      .map[Either[String, Option[Input]]] {
+        case None =>
+          val default = cwd.resolve(Configuration.Scalafix.defaultConfPath)
+          if (Files.isRegularFile(default.toNIO))
+            Right(Some(Input.File(default)))
+          else Right(None)
+        case Some(relpath) =>
+          val custom = cwd.resolve(relpath)
+          if (Files.isRegularFile(custom.toNIO))
+            Right(Some(Input.File(custom)))
+          else if (relpath == Configuration.Scalafix.defaultConfPath)
+            Right(None)
+          else
+            Left(s"metals.scalafix.confPath=$relpath is not a file")
+      }
+      .toFunction0()
+
   private def analyzeIndex(
       document: m.Document,
       index: SemanticdbIndex
-  ): Seq[Diagnostic] =
-    withConfig { configInput =>
-      val lazyIndex = lazySemanticdbIndex(index)
-      val configDecoder = ScalafixReflect.fromLazySemanticdbIndex(lazyIndex)
-      val (rule, config) =
-        ScalafixConfig.fromInput(configInput, lazyIndex)(configDecoder).get
-      val results: Seq[Diagnostic] = Parser.parse(document.input) match {
-        case Parsed.Error(_, _, _) => Nil
-        case Parsed.Success(tree) =>
-          val ctx = RuleCtx.applyInternal(tree, config)
-          val patches = rule.fixWithNameInternal(ctx)
-          Patch.lintMessagesInternal(patches, ctx).map(_.toLSP)
-      }
-
-      // megaCache needs to die, if we forget this we will read stale
-      // snapshots of filenames if using m.Input.File.slurp
-      // https://github.com/scalameta/scalameta/issues/1068
-      PlatformTokenizerCache.megaCache.clear()
+  ): Task[Either[Response.Error, Seq[Diagnostic]]] = Task {
```

Comment:
Can we keep the old signature? I would like to refactor lsp4s to make responses `Task[T]` instead of `Either[Response.Error, T]`. I've noticed that at least vsode doesn't display Response.Error responses unless they're server errors.

Reply:
We can't really keep the old signature, because now the implementation needs to read the configuration, which may fail. Would you prefer for this to be `Task[Seq[Diagnostic]]` where I make the `Task` fail if something goes wrong with the config?

Reply:
I don't think `Left(Response.Error)` responses are reported in at least vsode, which makes it better to showMessage or throw "internal error". Either way, we can keep it like this and later refactor the signature. The plan is to make it `Task[T]` with custom recovering logic for a handful of exceptions.

Reply:
Opened https://github.com/scalameta/metals/issues/206 to track this